shared/src/pool_fetching.rs

```diff
@@ -299,57 +315,81 @@ impl PoolFetching for PoolFetcher {
                 // Fetch ERC20 token balances of the pools to sanity check with reserves
                 let token0 = ERC20::at(&self.web3, pair.get().0);
                 let token1 = ERC20::at(&self.web3, pair.get().1);
-                (
-                    pair,
-                    pair_contract
-                        .get_reserves()
-                        .block(at_block.into())
-                        .batch_call(&mut batch),
-                    token0
-                        .balance_of(pair_address)
-                        .block(at_block.into())
-                        .batch_call(&mut batch),
-                    token1
-                        .balance_of(pair_address)
-                        .block(at_block.into())
-                        .batch_call(&mut batch),
-                )
+
+                let reserves = pair_contract
+                    .get_reserves()
+                    .block(at_block.into())
+                    .batch_call(&mut batch);
+                let token0_balance = token0
+                    .balance_of(pair_address)
+                    .block(at_block.into())
+                    .batch_call(&mut batch);
+                let token1_balance = token1
+                    .balance_of(pair_address)
+                    .block(at_block.into())
+                    .batch_call(&mut batch);
+
+                async move {
+                    // Clippy is wrong about this being eval order depndendent.
```

Comment:
The lints are probably there because of the `await`s.

I think its 100% fine to ignore in this case.